pkg/native/serializers/serializer_cdx.go

```diff
@@ -172,6 +185,7 @@ func (s *SerializerCDX) dependencies(ctx context.Context, bom *sbom.Document) ([
 	}
 
 	for _, e := range bom.NodeList.Edges {
+		e := e
```

Comment:
@houdini91 this is how u solve the "unsafe" memory in a loop.

Reply:
Thanks, I am interested can you elaborate on way this is safer?

Reply:
In each iteration, the iteration variable is assigned the value of the next element in the range expression. The variable itself doesn't change, only its value does. That's why the expression 'e' is pointing to the same location in memory throughout the loop. (this is why its unsafe)

This is why using 
`e := e` fixes this, because you create a new `e` variable in scope.